### ISSUE TITLE
New image source properties should not be blank.

### ISSUE THREAD

**Issue by Helioviewer-Kirill** [Helioviewer-Project/api#43]
In the KCOR image metadata the detector attribute key is set to a blank string, this breaks things.

`'measurement': '735', 'YCEN': 'NULL', 'scale': 5.643, 'instrument': u'COSMO', 'width': 1024, 'CROTA1': 'NULL', 'detector': '', 'NAXIS1': 1024, 'NAXIS2': 1024, 'RADIUS': 'NULL', 'date': datetime.datetime(2018, 11, 24, 21, 23, 25), 'height': 1024, 'nickname': '', 'CRPIX1': 512.5, 'observatory': u'COSMO_K-Coronagraph', 'CRVAL2': 0.0, 'XCEN': 'NULL', 'CRPIX2': 512.5, 'CRVAL1': 0.0, 'layeringOrder': 1`

If the field is blank it should not be included in the image metadata similar to SDO AIA

`'measurement': '171','YCEN': 0.0, 'scale': 0.6083709184772507, 'instrument': u'AIA', 'width': 4096, 'CROTA1': 'NULL', 'detector': 'AIA', 'NAXIS1': 4096, 'NAXIS2': 4096, 'RADIUS': 'NULL', 'date': datetime.datetime(2011, 6, 1, 0, 0, 0, 340000), 'height': 4096, 'nickname': 'AIA', 'CRPIX1': 2048.5, 'observatory': u'SDO', 'CRVAL2': 0.0, 'XCEN': 0.0, 'CRPIX2': 2048.5, 'CRVAL1': 0.0, 'layeringOrder': 1`

There is no detector field present in the AIA jp2 metadata.

**Comment by wafels** [comment]
I think the problem is that SunPy does not support KCor images.  This is an easy fix.... famous last words!

**Comment by wafels** [comment]
In this function

https://github.com/Helioviewer-Project/api/blob/master/install/helioviewer/jp2parser.py#L35

we rely on the data being understood by SunPy.  If SunPy does not know about the type of map data, then various parameters are set to default values.  The fix is to write a source class for KCor FITS and JP2 data in SunPy.

**Comment by Helioviewer-Kirill** [comment]
Thanks for the tip.  It now makes sense why we need to add a map file to sunpy as specified in the wiki.
[](http://wiki.helioviewer.org/wiki/Adding_Support_for_New_Data)

I followed through with this and after many attempts of trial and error I finally got it to work.

Here's what I found out:
The sunpy map file specifies the default values for image headers and can overwrite existing fields.
The datasource_property db table must contain the values specified in the image header or sunpy map.
